src/game/AI/ScriptDevAI/scripts/outland/hellfire_citadel/shattered_halls/shattered_hallsScripts.cpp

```diff
@@ -270,6 +269,251 @@ struct npc_shattered_hand_centurion : public CombatAI
+        if (pointId == 100)
+        {
+            summoned->GetMotionMaster()->MoveIdle();
+            summoned->SetInCombatWithZone();            
+        }
+    }
+
+    void CallForReinforcements()
+    {
+        // reinforcement can get spawned even if legionnaire is outfight and has a cooldown between 10 and 15 seconds, but only one can be up
+        if (!m_reinfCD)
+        {
+            if (m_creature->HasStringId(SECOND_LEGIONNAIRE_STRING) || m_creature->HasStringId(THIRD_LEGIONNAIRE_STRING) || m_creature->HasStringId(FOURTH_LEGIONNAIRE_STRING) || m_creature->HasStringId(FIFTH_LEGIONNAIRE_STRING))
+            {
+                if (Creature* felorc = m_creature->SummonCreature(MOB_FEL_ORC, FelOrcSpawnCoords[legionnaireGuid][0], FelOrcSpawnCoords[legionnaireGuid][1], FelOrcSpawnCoords[legionnaireGuid][2], FelOrcSpawnCoords[legionnaireGuid][3], TEMPSPAWN_TIMED_OOC_OR_DEAD_DESPAWN, urand(20000, 25000), true, true))
+                {
+                    felorc->GetMotionMaster()->MoveWaypoint(legionnaireGuid, 0, 0, 0, FORCED_MOVEMENT_RUN);
+                    felorc->SetCanCallForAssistance(false);
+                    felorc->SetCanCheckForHelp(false);
+                    DoBroadcastText(aRandomReinf[urand(0, 6)], m_creature);
+                }
+            }
+            m_reinfCD = true;
+        }
+
+        // Legionnaire 006
+        if (m_creature->HasStringId(SIX_LEGIONNAIRE_STRING))
+        {
+            // there are 4 sleeping npcs around him, if one of his group members dies he will call for one of the sleeping creatures to get up and join the fight
+            // this doesnt have a cd, if all 4 npcs with sleeping aura are up, nothing more happens
+            auto m_sleepingReinf = m_creature->GetMap()->GetCreatures(SLEEPING_REINF_STRING);
+            Creature* closest = nullptr;
+            for (Creature* creature : *m_sleepingReinf)
+            {
+                // Only call alive creatures
+                // Only call creature that isnt in combat already
+                // Only call creature that still has sleeping aura
+                if (creature->IsAlive() && !creature->IsInCombat() && !creature->HasAura(AURA_SLEEPING))
+                    continue;
+
+                if (!closest)
+                    closest = creature;
+                else if (m_creature->GetDistance(creature, true, DIST_CALC_NONE) < m_creature->GetDistance(closest, true, DIST_CALC_NONE))
+                    closest = creature;
+            }
+            if (closest)
+            {
+                DoBroadcastText(aRandomReinfSleeping[urand(0, 6)], m_creature);
+                closest->RemoveAurasDueToSpell(AURA_SLEEPING);
+                closest->SetIgnoreMMAP(true); // hackfix
+                closest->GetMotionMaster()->MovePoint(100, FelOrcSpawnCoords[legionnaireGuid][0], FelOrcSpawnCoords[legionnaireGuid][1], FelOrcSpawnCoords[legionnaireGuid][2], FORCED_MOVEMENT_RUN);
```

Comment:
I think you can remove the hackfix here. MovePoint has an optional param GeneratePath which if you set to false, will ignore pathfinding and just straight up move to the destination. That way once mob gets into combat, it will still use mmaps as intended.